rsocket-core/src/main/java/io/rsocket/transport/Transport.java

```diff
@@ -16,5 +16,22 @@
 
 package io.rsocket.transport;
 
+import static io.rsocket.frame.FrameLengthCodec.FRAME_LENGTH_MASK;
+
+import io.rsocket.DuplexConnection;
+
 /** */
-public interface Transport {}
+public interface Transport {
+
+  /**
+   * Configurations that exposes the maximum frame size that a {@link DuplexConnection} can bring up
+   * to RSocket level.
+   *
+   * <p>This number should not exist the 16,777,215 (maximum frame size specified by RSocket spec)
+   *
+   * @return return maximum configured frame size limit
+   */
+  default int maxFrameLength() {
```

Comment:
moved maxFrameLength property to the transport, since it describes transport behavior and from my point of view checking that when a connection has already wired is too late

Reply:
Also, this one is a default property, so it should not be a breaking change for anybody